File: 023. Run-Length Encoding/RunLengthEncoding.java
import java.util.*;

class RunLengthEncoding {
    public String runLengthEncoding(String string) {
        int currentRunLength = 1;
        StringBuilder encodedString = new StringBuilder();
        char currentChar = string.charAt(0);
        for (int i = 1; i < string.length(); i++) {
            if ((string.charAt(i) != currentChar) || (currentRunLength == 9)) {
                encodedString.append(currentRunLength);
                encodedString.append(currentChar);
                currentRunLength = 0;
                currentChar = string.charAt(i);
            }
            currentRunLength++;
        }
        encodedString.append(currentRunLength);
        encodedString.append(string.charAt(string.length() - 1));
        return encodedString.toString();
    }
}

File: 023. Run-Length Encoding/RunLengthEncoding.kt
package com.algoexpert.program

fun runLengthEncoding(string: String): String {
    var currentRunLength = 1
    val encodedString = mutableListOf<Char>()
    var currentChar = string[0]
    for (i in 1 until string.length) {
        if ((string[i] != currentChar) || (currentRunLength == 9)) {
            encodedString.add(currentRunLength.toString()[0])
            encodedString.add(currentChar)
            currentRunLength = 0
            currentChar = string[i]
        }
        currentRunLength++
    }
    encodedString.add(currentRunLength.toString()[0])
    encodedString.add(string[string.length - 1])
    return encodedString.joinToString("")
}
